lib/SIL/IR/SILConstants.cpp

```diff
@@ -1027,7 +1028,7 @@ void SymbolicValue::emitUnknownDiagnosticNotes(SILLocation fallbackLoc) {
     getWitnessMethodName(dyn_cast<WitnessMethodInst>(unknownNode),
                          witnessMethodName);
     diagnose(ctx, diagLoc, diag::constexpr_unresolvable_witness_call,
-             witnessMethodName);
+             std::string(witnessMethodName.c_str()));
```

Comment:
The std::string conversions here and elsewhere shouldn't be necessary, the argument will still be passed as a `StringRef` and the diagnostics engine will make its own copy of any string that needs to outlive the call to `diagnose`.

Reply:
I see. That is good to know. The place where things are getting copied wasn't very obvious.

Reply:
I will clean it up in the next commit.

Reply:
👍 . You're right that it's really not clear we have special handling for string arguments to diagnostics. The copy happens in `onTentativeDiagnosticFlush` at the bottom of `DiagnosticEngine.cpp`, which is called when the destructor of the `InFlightDiagnostic` returned by `diagnose` runs. I'll make a note to see if I can find somewhere to document this behavior.

Reply:
Sounds great. That would be very helpful.